dome9/data_source_dome9_users.go

```diff
@@ -0,0 +1,296 @@
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"calculated_permissions": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"access": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"manage": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"rulesets": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"notifications": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"policies": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"alert_actions": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"create": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"view": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"on_boarding": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"cross_account_access": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"is_mobile_device_paired": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceUsersRead(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+
+	id := d.Get("id").(string)
+	log.Printf("[INFO] Getting data user with id %s\n", id)
+
+	resp, _, err := d9Client.users.Get(id)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.Itoa(resp.ID))
+	_ = d.Set("email", resp.Name)
+	_ = d.Set("is_sso_enabled", resp.SsoEnabled)
+	_ = d.Set("is_suspended", resp.IsSuspended)
+	_ = d.Set("is_owner", resp.IsOwner)
+	_ = d.Set("is_super_user", resp.IsSuperUser)
+	_ = d.Set("is_auditor", resp.IsAuditor)
+	_ = d.Set("has_api_key", resp.HasAPIKey)
+	_ = d.Set("has_api_key_v1", resp.HasAPIKeyV1)
+	_ = d.Set("has_api_key_v2", resp.HasAPIKeyV2)
+	_ = d.Set("is_mfa_enabled", resp.IsMfaEnabled)
+	_ = d.Set("role_ids", resp.RoleIds)
+	_ = d.Set("iam_safe", flattenIamSafe(resp.IamSafe))
+	_ = d.Set("can_switch_role", resp.CanSwitchRole)
+	_ = d.Set("is_locked", resp.IsLocked)
+	_ = d.Set("last_login", resp.LastLogin.Format("2006-01-02 15:04:05"))
+	_ = d.Set("permissions", flattenPermissions(resp.Permissions))
```

Comment:
you use `flattenPermissions(resp.Permissions)` for both `permissions` and `calculated_permissions`. Is it ok?